src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -1,26 +1,59 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
 using System.CommandLine.Parsing;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
 using Microsoft.DotNet.Cli;
 using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.Workloads.Workload.Install;
 
 namespace Microsoft.DotNet.Workloads.Workload.List
 {
     internal class WorkloadListCommand : CommandBase
     {
+        private readonly string[] _machineReadableOption;
+
+        public readonly string MockInstallDirectory = Path.Combine(CliFolderPathCalculator.DotnetUserProfileFolderPath,
+            "DEV_mockworkloads");
+
         public WorkloadListCommand(
             ParseResult result
         )
             : base(result)
         {
-  
+            _machineReadableOption = result.ValueForOption<string[]>(WorkloadListCommandParser.MachineReadableOption);
         }
 
         public override int Execute()
         {
             // TODO stub
-            Reporter.Output.Write("WIP stub workload list");
+            var installedList = new List<string>();
```

Comment:
Should all of this be behind a check if `_machineReadableOption` is null/empty (in case the user didn't specify it)?

Reply:
Add to the list https://github.com/dotnet/sdk/issues/16484#issue-838941792

Reply:
changed _machineReadableOption to a bool